src/Microsoft.DotNet.Build.Tasks/ExecWithRetriesForNuGetPush.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Run a command and retry if the exit code is not 0.
+    /// </summary>
+    public class ExecWithRetriesForNuGetPush : ExecWithRetries
```

Comment:
Seems weird to derive from `ExecWithRetries` then only use the input properties and break the `ICancelableTask` implementation. If you want to avoid refactoring `ExecWithRetries` because it's a quick FR thing (understandable) I think I'd actually prefer if this were a pure dupe. (With a TODO pointing to a buildtools issue.)